docker/Dockerfile.dev

```diff
@@ -0,0 +1,91 @@
+FROM python:3.8-slim as base
```

Comment:
Should we grab a new Python image at some point?

Reply:
maybe. I wouldn't mind features of 3.9, but 3.8 is plenty modern for now